flask/app/buckets.py

```diff
@@ -33,3 +39,46 @@ def get_object_info():
             )
 
     return jsonify(all_objects)
+
+
+@app.route("/api/unlinked", methods=["GET"])
+@login_required
+def get_unlinked_files():
+
+    # Get all minio bucket names
+    all_bucket_names = [bucket.name for bucket in minioClient.list_buckets()]
+
+    # Remove buckets the current user does not have access to assuming group_code.lower() == bucket name
+    user = (
+        db.session.query(models.User)
+        .filter_by(user_id=current_user.user_id)
+        .options(joinedload(models.User.groups))
+        .first()
+    )
+
+    valid_bucket_names = []
+    for group in user.groups:
+        code = group.group_code.lower()
+        if code in all_bucket_names:
+            valid_bucket_names.append(code)
+
+    # Get all files in valid minio buckets
+    all_files = []
+    for bucket in valid_bucket_names:
+        objs = minioClient.list_objects(bucket)
+        for obj in objs:
+            all_files.append(bucket + "/" + obj.object_name)
+
+    # Get all linked files
+    linked_files = {}
+    for dataset in models.Dataset.query.all():
+        if dataset.input_hpf_path is not None:
+            linked_files[dataset.input_hpf_path] = ":)"
+
+    # Put all unlinked files in new list
+    unlinked_files = []
+    for file_name in all_files:
+        if file_name[file_name.index('/')+1:] not in linked_files:
```

Comment:
This should just be `file_name` because we should be storing the combined bucket and object name that is unique

Reply:
oh so `input_hpf_path` will have the bucket prefix as well, got it